test/sqllogictest/explain.slt

```diff
@@ -108,7 +108,7 @@ EXPLAIN TYPED RAW PLAN FOR SELECT (SELECT column2), (SELECT (SELECT column1)) FR
 | | | | | | types = ()
 | | | | | | keys = ()
 | | | | | Map #^^0
-| | | | | | types = (integer)
+| | | | | | types = (integer?)
```

Comment:
It turns out to be a harmless bug in `explain.rs`. It just needs to catch up with the order change. Something like this would fix it:

```
$ git diff
diff --git a/src/sql/src/plan/explain.rs b/src/sql/src/plan/explain.rs
index 55d081061..287e4240b 100644
--- a/src/sql/src/plan/explain.rs
+++ b/src/sql/src/plan/explain.rs
@@ -229,11 +229,11 @@ impl<'a> Explanation<'a> {
             // TODO(jamii) `typ` is itself recursive, so this is quadratic :(
             let typ = node.expr.typ(outers, params);
             let mut outers = outers.to_vec();
-            outers.push(typ);
+            outers.insert(0, typ);
             for subquery in &mut node.subqueries {
                 subquery.explain_types_internal(&outers, params);
             }
-            node.typ = outers.pop();
+            node.typ = outers.drain(0..1).next();
         }
     }
 

```

Reply:
Note that this issue leads to a crash when the number of columns don't match:

```
materialize=> create table t(column1 int, column2 string);
CREATE TABLE
materialize=> explain typed raw plan for SELECT (SELECT (SELECT column2)::int) FROM t;
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
!> 

```

Reply:
Whoooops! 🙈 Thanks for not letting me get away with that!